src/components/Dialogs/LayoutDialog/SaveLayoutDialogComponent.tsx

```diff
@@ -54,13 +54,24 @@ export class SaveLayoutDialogComponent extends React.Component {
         this.clearInput();
     };
 
+    private renameLayout = async () => {
+        const appStore = AppStore.Instance;
+        await appStore.layoutStore.renameLayout(appStore.layoutStore.oldLayoutName, this.layoutName.trim());
+        this.clearInput();
+    };
+
     @computed get isEmpty(): boolean {
-        return !this.layoutName;
+        return !this.layoutName?.trim();
+    }
+
+    @computed get validName(): boolean {
+        return this.layoutName.match(/^[^~`!*()-+=[.'?<>/|\\:;&]+$/)?.length > 0;
```

Comment:
@TienHao could you check why the minus sign "-" is _not_ detected as invalid char?